### ISSUE TITLE
XQuery parser does not recognize node constructors as final step of an XPath

### ISSUE THREAD

**Issue by ronhitchens** [overstory/marklogic-intellij-plugin#49]
Expressions like this result in a syntax error, even though they are valid XQuery 3.0:

/foo/bar/baz/element myelement { ./data() }

They can, however, be worked around by enclosing the node constructor in parenthesis:

/foo/bar/baz/(element myelement { ./data() })

**Comment by rhdunn** [comment]
The `element "myelement" {...}` syntax is invalid, it should either be `element myelement {...}` or `element {"myelement"} {...}` as `CompElemConstructor` is defined as:

    "element" (EQName | ("{" Expr "}")) "{" ContentExpr? "}"

Reference: https://www.w3.org/TR/xquery-30/#prod-xquery30-CompElemConstructor (XQuery 3.0 EBNF [153]).

**Comment by ronhitchens** [comment]
Yes, you're right.  I typed that in by hand.  I've edited the original.
Thanks.
